Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -159,28 +159,33 @@ extension Array: BsonValue {
 }
 
 /// Subtypes for BSON Binary values.
-public enum BsonSubtype: Int {
+public enum BsonSubtype: Int, Decodable {
     /// Generic binary subtype
-    case binary = 0x00,
+    case binary = 0,
     /// A function
-    function = 0x01,
+    function,
     /// Binary (old)
-    binaryDeprecated = 0x02,
+    binaryDeprecated,
     /// UUID (old)
-    uuidDeprecated = 0x03,
+    uuidDeprecated,
     /// UUID
-    uuid = 0x04,
+    uuid,
     /// MD5
-    md5 = 0x05,
+    md5,
     /// User defined
-    user = 0x06
+    user
 }
 
 /// A struct to represent the BSON Binary type.
-public struct Binary: BsonValue, Equatable {
+public struct Binary: BsonValue, Equatable, Decodable {
+
     public var bsonType: BsonType { return .binary }
-    var data: Data
-    var subtype: BsonSubtype
+
+    /// The binary data.
+    public let data: Data
```

Comment:
bunch of properties in our `BsonValue` structs were not public but should be, oops 😳

Reply:
we keep finding more of these! I wonder if it's possible to make a sample program that we ship with the driver that actively uses all types from outside the package in order to catch issues like this earlier? Or are there any linting tools we can use for this purpose?

Reply:
Hmm, I'll look into it... Now that we have docs (almost), one easy thing to do is to look at the generated docs for a type and see if the properties and methods you expect show up there since docs are only generated for public stuff.

Reply:
:+1: